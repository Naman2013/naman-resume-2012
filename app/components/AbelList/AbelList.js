import React from 'react';
import PropTypes from 'prop-types';

const AbelList = ({ list }) => (
  <ul>
    { list.map(item => <li>{item}</li>) }
  </ul>
);

AbelList.propTypes = {
  list: PropTypes.arrayOf(PropTypes.string),
};

AbelList.defaultProps = {
  list: [],
};

export default AbelList;
