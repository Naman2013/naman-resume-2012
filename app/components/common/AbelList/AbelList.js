import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { DeviceContext } from '../../../providers/DeviceProvider';
import style from './AbelList.style';

const AbelList = ({ list }) => (
  <Fragment>
    <DeviceContext.Consumer>
      {
        (context) => {
          console.log(context);
          return (
            <ul>
              { list.map(item => <li>{item}</li>) }
            </ul>
          );
        }
      }

    </DeviceContext.Consumer>
    <style jsx>{style}</style>
  </Fragment>
);

AbelList.propTypes = {
  list: PropTypes.arrayOf(PropTypes.string),
};

AbelList.defaultProps = {
  list: [],
};

export default AbelList;
