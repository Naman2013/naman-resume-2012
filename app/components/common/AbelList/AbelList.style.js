import css from 'styled-jsx/css';
import { primaryFont } from '../../../styles/variables/fonts';

export default css`
  ul {
    font-family: ${primaryFont};
    list-style-type: none;
    margin: 0;
    padding: 0;
  }

  li {
    border-bottom: 1px solid aqua;
  }
`;
