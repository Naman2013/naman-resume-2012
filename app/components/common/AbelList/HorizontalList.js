import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import style from './HorizontalList.style';

const HorizontalList = ({ list, theme, iconList }) => (
  <Fragment>
    <ul style={theme}>
      {list.map((entry, i) => (
        <li key={`horizontal-list-${entry}`}>
          {' '}
          {iconList[i] && <img className="bullet-image" src={iconList[i]} />}
          {entry}
        </li>
      ))}
    </ul>
    <style jsx>{style}</style>
  </Fragment>
);

HorizontalList.propTypes = {
  list: PropTypes.arrayOf(PropTypes.string).isRequired,
  theme: PropTypes.shape({}),
  iconList: PropTypes.arrayOf(PropTypes.string),
};

HorizontalList.defaultProps = {
  theme: {},
  iconList: [],
};

export default HorizontalList;
