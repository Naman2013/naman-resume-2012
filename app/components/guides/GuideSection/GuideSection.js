import React from 'react';
import PropTypes from 'prop-types';
import noop from 'lodash/noop';

const GuideSection = ({ content, column, contentAlign }) => (
  <div />
);

GuideSection.propTypes = {
  content: PropTypes.string.isRequired,
  column: PropTypes.func,
  alignContent: PropTypes.oneOf(['left', 'right']),
};

GuideSection.defaultProps = {
  column: noop,
  alignContent: 'left',
};

export default GuideSection;
