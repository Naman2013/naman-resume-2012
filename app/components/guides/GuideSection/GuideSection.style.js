import css from 'styled-jsx/css';

export default css`
  .content-container, .column-container {
    margin: 0;
    padding: 0;
  }
`;
