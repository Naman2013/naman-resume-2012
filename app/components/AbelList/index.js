import AbelList from './AbelList';

export default AbelList;
