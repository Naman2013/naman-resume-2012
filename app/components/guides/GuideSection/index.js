import GuideSection from './GuideSection';

export default GuideSection;
