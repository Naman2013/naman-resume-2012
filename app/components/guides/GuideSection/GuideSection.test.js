import React from 'react';
import { shallow } from 'enzyme';
import AbelList from '../../common/AbelList';
import GuideSection from './GuideSection';

describe('GuideSection', () => {
  const props = {
    content: `Mauris non tempor quam, et lacinia sapien. Mauris accumsan eros eget libero lorem posuere vulputate. Etiam elit elit, elementum sed varius at, adipiscing evitae est. Sed nec felis loren posuere posuere, rutrum eu ipsum. Aliquam eget odio sed ligula dolae iaculis consequat at eget orci. Mauris moleistie sit amet metus loi mass imattis varius Donec sit amet ligula eget nisi sodales lorem a molestie bibendum. Etiam nisi anteni consectetur eget placerat a, tempus a neque. Donec ut elit urna. Etiam venenatis leni eleifend urna eget scelerisqueliquam in nunc.

Donec sit amet ligula eget nisi sodales egestas. Aliquam interdum dolor aliquet dolor  iaculis consequat at eget orci. Mauris moleistie sit amet metus loi mass imattis varius Donec sit amet ligula eget nisi sodales lorem a molestie bibendum. Etiam nisi anteni
posuere vulputate. Etiam elit elit, elementum sed varius at.`,
    column: () => (<AbelList list={['contains 10 objects', 'object type guide', 'beginners and up']} />),
    contentAlign: 'left',
  };

  const shallowWrapper = shallow(<GuideSection {...props} />);
  it('should render correctly', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });
});
