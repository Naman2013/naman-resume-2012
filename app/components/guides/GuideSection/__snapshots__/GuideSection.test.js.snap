// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuideSection should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GuideSection
    alignContent="left"
    column={[Function]}
    content="Mauris non tempor quam, et lacinia sapien. Mauris accumsan eros eget libero lorem posuere vulputate. Etiam elit elit, elementum sed varius at, adipiscing evitae est. Sed nec felis loren posuere posuere, rutrum eu ipsum. Aliquam eget odio sed ligula dolae iaculis consequat at eget orci. Mauris moleistie sit amet metus loi mass imattis varius Donec sit amet ligula eget nisi sodales lorem a molestie bibendum. Etiam nisi anteni consectetur eget placerat a, tempus a neque. Donec ut elit urna. Etiam venenatis leni eleifend urna eget scelerisqueliquam in nunc.

Donec sit amet ligula eget nisi sodales egestas. Aliquam interdum dolor aliquet dolor  iaculis consequat at eget orci. Mauris moleistie sit amet metus loi mass imattis varius Donec sit amet ligula eget nisi sodales lorem a molestie bibendum. Etiam nisi anteni
posuere vulputate. Etiam elit elit, elementum sed varius at."
    contentAlign="left"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
